### PR TITLE
Implementing new work design and groupings

### PR DESCRIPTION
**Pull request by gestchild**
References #3954

- This is currently a bit of a hybrid. I've kept the 'Downloads',  'Using this ...' and 'We're improving..." sections (but applied the new layout to them)
- Where there are discrepancies in the labels between the design file and @jennpb 's sketch in #3954 I've used the latter 
- Still awaiting tag design to be finalised for subjects
- N.B. none of the data that's moved to the new header is now in the page

localhost_3001_works_evzcgfjz_query genomics worktype a items locations locationtype iiif-image 2ciiif-presentation

localhost_3001_works_kjwstxzy_query optics

### CODE REVIEW

**Comment by jamesgorrie** [comment]
> * Still awaiting tag design to be finalised for subjects

Assuming this doesn't need to be done now.

We could be waiting a while on that, where the focus now is on this, and we're almost there for releasing.

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js [hunk 1]
```diff
@@ -385,106 +201,188 @@ const WorkDetails = ({
...
-            </SpacingComponent>
+            </WorkDetailsSection>
           </div>
         </div>
       </div>
     </div>
   );
 };
 
+/* TODO old stuff without a home below here
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: Can we just make a decision about this?

**Comment by gestchild** [comment]
> > * Still awaiting tag design to be finalised for subjects
> 
> Assuming this doesn't need to be done now.
> 
> We could be waiting a while on that, where the focus now is on this, and we're almost there for releasing.

Yeah, wasn't suggesting we wait - just a nod to why it doesn't look like the design file

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: Yeah, I'll chat with @jennpb

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js [hunk 2]
```diff
@@ -385,106 +201,188 @@ const WorkDetails = ({
...
+                {work.language && (
+                  <MetaUnit
+                    headingLevel={3}
+                    headingText="Language"
+                    links={[
+                      <NextLink
+                        key={1}
+                        {...worksUrl({
+                          query: `"${work.language.label}"`,
+                          page: undefined,
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: 👍

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js [hunk 3]
```diff
@@ -44,294 +142,11 @@ const WorkDetails = ({
...
-                <MoreLink
-                  name="View Wellcome Library catalogue record"
-                  url={encoreLink}
-                />
-              </SpacingComponent>
-            )}
             {iiifImageLocationUrl && (
-              <SpacingComponent>
+              <WorkDetailsSection
+                showDivider={false}
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: Perhaps, I put it inside, so didn't have to repeat the logic for showing the divider and showing the WorkDetailsSection - I'm happy to change it though if it makes more sense to you that way

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js [hunk 4]
```diff
@@ -1,22 +1,96 @@
...
+  .work-details-body {
+    grid-column: 1 / -1;
+  }
+
+  @media (min-width: 800px) {
+    .work-details-heading {
+      grid-column: span 4;
+    }
+
+    .work-details-body {
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: Yeah, I'm making use of work @davidpmccormick did on the initial prototype (MetaUnit2 component). But if we're all happy I think it makes sense to move all the styles into styled components. Maybe not this PR though.

File: common/views/components/Buttons/Button/Button.js [hunk 5]
```diff
@@ -62,7 +62,9 @@ const Button = ({
     >
       <span className="flex-inline flex--v-center">
         {icon && <Icon name={icon} />}
-        <span className="btn__text">{text}</span>
+        <span className={`btn__text ${spacing({ s: 0 }, { margin: ['top'] })}`}>
```
Comment (jamesgorrie) [inline common/views/components/Buttons/Button/Button.js]: Why does this have margin on the top?
Could we use `classNames({})` if we're adding more than 1 class?

File: catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js [hunk 6]
```diff
@@ -34,8 +108,32 @@ const WorkDetails = ({
   iiifImageLocationLicenseId,
   encoreLink,
 }: Props) => {
+  const singularWorkTypeLabel = work.workType.label
+    ? work.workType.label.replace(/s$/g, '').toLowerCase()
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply catalogue/webapp/components/WorkDetails/WorkDetailsNewDataGrouping.js]: WONT DO as we only use this in s sentence like structure.

**Pull request merged**